bundles/org.openhab.binding.modbus/src/main/java/org/openhab/binding/modbus/handler/BaseModbusThingHandler.java

```diff
@@ -59,9 +58,27 @@ public BaseModbusThingHandler(Thing thing) {
     public void initialize() {
```

Comment:
Hmm this will now raise unchecked exception when bridge is offline e.g due to configuration error? When bridge is having configuration error, the slave id is not available.

Perhaps the configuration error of bridge is the only situation when these methods can fail?

Should it set thing to offline with BRIDGE_OFFLINE instead of raising unchecked exception to framework?

Reply:
I see no other chance to prevent the Thing from being initialized. Setting the Thing status has no effect as it will be overwritten by any `updateStatus()` in `initialize()` of the "user" class. Do you have a suggestion?

Reply:
Good point... 

I looked at some examples from the code base

- `AbstractModbusEndpointThingHandler` parent class sets thing OFFLINE. initialize not implemented in child class
- `BeaconBluetoothHandler` (in bluetooth) parent class sets thing OFFLINE. Child class (RuuviTagHandler) has check 
```
        if (getThing().getStatus() != ThingStatus.OFFLINE) {
            heartbeatFuture = scheduler.scheduleWithFixedDelay(this::heartbeat, 0, HEARTBEAT_TIMEOUT_MINUTES,
                    TimeUnit.MINUTES);
```
I there might be race condition here... Ist getThing().getStatus() update instantaneous?
- `IT100BridgeHandler extends DSCAlarmBaseBridgeHandler` super initialize is called conditionally when config is otherwise OK.

Do we typically expect to override the initialize method in child class?

Reply:
It's always good to have a second pair of eyes. The examples inspired me to come up with a slightly other solution.

The BaseModbusThingHandler got a new abstract method `modbusInitialize()`, which is only called, when the Modbus Bridge is configured correctly. That way the Thing can be set to offline in a normal way. Also, the user doesn't need to call `super.initialize()` anymore, which was error-prone. So, the boolean flag could be dropped, too. I made the `initialize()` method `final` to prevent the user from overriding it wrongly.